src/MUIDataTable.js

```diff
@@ -259,52 +259,56 @@ class MUIDataTable extends React.Component {
     },
   };
 
-  state = {
-    activeColumn: null,
-    announceText: null,
-    count: 0,
-    columns: [],
-    expandedRows: {
-      data: [],
-      lookup: {},
-    },
-    data: [],
-    displayData: [],
-    filterData: [],
-    filterList: [],
-    page: 0,
-    previousSelectedRow: null,
-    rowsPerPage: 0,
-    searchProps: {},
-    searchText: null,
-    selectedRows: {
-      data: [],
-      lookup: {},
-    },
-    showResponsive: false,
-    sortOrder: {},
-  };
-
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.tableRef = React.createRef();
     this.tableContent = React.createRef();
     this.draggableHeadCellRefs = {};
     this.resizeHeadCellRefs = {};
     this.timers = {};
     this.setHeadResizeable = () => {};
     this.updateDividers = () => {};
-  }
 
-  UNSAFE_componentWillMount() {
-    this.initializeTable(this.props);
```

Comment:
initializeTable has changed the position of componentDidMount.

Reply:
Is this an issue?

Reply:
Nope,
I left a trace while reading the code. Sorry. 🙏